### DISCUSSION TITLE
Sorting results with a formula

### DISCUSSION THREAD

**Discussion by MrBadou** [microsoft-search/pnp-modern-search#1792] [Q&A]
If i'm not mistaken it is not possible to sort the results inside the results web part through a formula.

As it is said here :
https://docs.microsoft.com/en-us/sharepoint/dev/general-development/sorting-search-results-in-sharepoint
It is possible to do a search with a formula inside the sortlist or in the result web part i can only sort by managed properties.

I would like to do something like
_api/search/query?querytext=%27RefinableInt09:21005..21100%27&sortlist=%27[formula:abs(21100-RefinableInt11))]:ascending%27
Which give me a result if i use directly the API.

Thanks

**Comment by gretchunkim** [comment]
@MrBadou Is Is this using SharePoint or Microsoft Search as data source?  In Microsoft Search as a data source, sorting is limited for externalItems, an I do see mixed documentation of whether this is the case or not.

**Reply by MrBadou** [reply]
@gretchunkim Hello, thank you for your answer
It is using SharePoint as data source.
It was some time ago and i managed to do what i wanted to do in another way.

**Reply by gretchunkim** [reply]
@MrBadou You can sort SharePoint as data source. Instead of using formula, you can use "sort settings" in 1 of 4 of the PnP - Search Results.

![image](https://github.com/microsoft-search/pnp-modern-search/assets/8564035/b316b6ed-0f15-4982-ab65-b1d99a1252c3)

**Reply by MrBadou** [reply]
No i couldn't use that because i needed to use a mathematical formula using many field to sort

**Comment by kasperbolarsen** [comment]
That only workaround I know of that can solve that is to use search verticals, and each of the results web parts are using a different ranking using XRANK
https://www.techmikael.com/2016/12/tip-for-writing-multiple-xranks-or.html

Could that be an option ?

**Reply by MrBadou** [reply]
Hi,
It's generous of you to answer but i already found a workaround since i posted this question in February of 2022.
Thank you

**Reply by wobba** [reply]
@MrBadou can you share your workaround? :)
